packages/plugins/quorum/src/quorumWeb3Extensions.ts

```diff
@@ -0,0 +1,184 @@
+import { Embark } from "embark-core";
+import Web3 from "web3";
+import { AbiCoder as EthersAbi } from 'ethers/utils/abi-coder';
+import quorumjs from "quorum-js";
+
+// Create an augmentation for "web3-utils
+declare module "web3-utils" {
+  // Augment the 'web3-utils' class definition with interface merging
+  interface Utils {
+    _: any;
+    BN: any;
+  }
+}
+
+export class QuorumWeb3Extensions {
+
+  constructor(private embark: Embark) {
+  }
+
+  registerOverrides() {
+    return new Promise((resolve) => {
+      this.embark.registerActionForEvent<Web3>("runcode:register:web3", (web3: Web3, cb) => {
+        web3 = extendWeb3(web3);
+        cb(null, web3);
+        resolve();
+      });
+    });
+  }
```

Comment:
Just a thought: It looks like we're mutating `web3`. Would it make sense to considering creating a new object/clone + extending it, instead of mutating the one that's potentially being used in places where overridden APIs aren't expected (if such a case is possible)?

Reply:
On the contrary, why would users **not** want the original `web3` mutated? Isn't that the point of the method?

Reply:
Yes, good point. After our team call this was getting more clear to me too.

I guess there's a difference between extending something and overwriting something. Not sure if `extend()` is meant to be used for both cases.

Anyways, was just thinking out loud :)